File: Proyecto final tramo 3/src/static/js/courses.js
const apiUrl = 'https://potrero-digital-api.vercel.app/courses';

document.addEventListener('DOMContentLoaded', () => {
    fetchCourses();
});

function fetchCourses() {
    fetch(apiUrl)
        .then(response => response.json())
        .then(data => {
            displayCourses(data);
        })
        .catch(error => console.error('Error fetching courses:', error));
}

function displayCourses(courses) {
    const coursesContainer = document.getElementById('courses-container');
    coursesContainer.innerHTML = '';

    courses.forEach(course => {
        const courseCard = document.createElement('div');
        courseCard.className = 'course-card';
        courseCard.innerHTML = `
            <h3 class="course-title">${course.title}</h3>
            <p class="course-description">${course.description}</p>
            <button class="detail-button" onclick="showCourseDetail('${course.id}')">Ver Detalle</button>
        `;
        coursesContainer.appendChild(courseCard);
    });
}

function showCourseDetail(courseId) {
    // redirige a la página de detalle del curso
    window.location.href = `/detail/${courseId}`;
}

File: src/views/courses.py
from flask import Blueprint, render_template
import requests

courses_bp = Blueprint('courses', __name__)

@courses_bp.route('/courses')
def courses():
    api_url = 'https://potrero-digital-api.vercel.app/courses'
    response = requests.get(api_url)
    courses_data = response.json() if response.status_code == 200 else []

    return render_template('courses.html', courses=courses_data)